views/js/controller/Delivery/monitoring.js

```diff
@@ -176,1115 +178,1248 @@ define([
+                        dataPolling.stop();
+                    }
+
+                    if (deliveryId) {
+                        serviceParams.delivery = deliveryId;
+                    }
+                    if (context) {
+                        serviceParams.context = context;
                     }
 
-                    // tools: handles the session time
-                    if (timeHandlingMode) {
+                    return proxyExecutions.read(serviceParams).then(function (data) {
+                        dataset = data.set;
+                        extraFields = data.extrafields;
+                        categories = data.categories;
+                        deliveryId = data.delivery || deliveryId;
+                        context = data.context || context;
+                        timeHandlingMode = data.timeHandling === 'extra_time';
+                        changeTimeMode = data.timeHandling === 'timer_adjustment';
+                        allowedConnectivity = data.onlineStatus || false;
+                        printReportButton = data.printReportButton;
+                        printReportUrl = data.printReportUrl;
+                        hasAccessToReactivate = data.hasAccessToReactivate;
+                        sessionsHistoryUrl = data.historyUrl || historyUrl;
+                        dialogSettings = data.dialogSettings;
+
+                        var showColumnFirstName = extractOption(data, 'showColumnFirstName', true);
+                        var showColumnLastName = extractOption(data, 'showColumnLastName', true);
+                        var showColumnAuthorize = extractOption(data, 'showColumnAuthorize', true);
+                        var showColumnRemainingTime = extractOption(data, 'showColumnRemainingTime', true);
+                        var showColumnExtendedTime = extractOption(data, 'showColumnExtendedTime', true);
+                        var showActionShowHistory = extractOption(data, 'showActionShowHistory', true);
+                        var setStartDataOneDay = extractOption(data, 'setStartDataOneDay', true);
+
+                        if (deliveryId) {
+                            serviceParams.delivery = deliveryId;
+                        }
+                        if (context) {
+                            serviceParams.context = context;
+                        }
+
+                        /**
+                         * configurable parameter to show button
+                         */
+                        if (data.refreshBtn) {
+                            // tool: page refresh
+                            tools.push({
+                                id: 'refresh',
+                                icon: 'reset',
+                                title: __('Refresh the page'),
+                                label: __('Refresh'),
+                                action() {
+                                    $list.datatable('refresh', {});
```

Comment:
The `{}` param prevents the datatable to query the server, so that the grid will be emptied.

Reply:
Thank you. That were for exploring, forgot to clean it 😬